src/service/viewer-impl.js

```diff
@@ -276,6 +276,16 @@ export class Viewer {
     this.performanceTracking_ = this.params_['csi'] === '1';
     dev.fine(TAG_, '- performanceTracking:', this.performanceTracking_);
 
+    /**
+     * Whether the AMP document is embedded in a viewer, such as an iframe or
+     * a web view.
+     * Notice that the value of the "origin" parameter itself is never consulted
+     * here, but instead confirmed via handshake in the `setMessageDeliverer`
+     * method.
+     * @private @const {boolean}
+     */
+    this.isEmbedded_ = this.isIframed_ || !!this.params_['origin'];
```

Comment:
No, I meant to add a method of that name that does whatever work it needs to do. I don't really see how it follows out of the origin being present.